### PR TITLE
CI: add image scanning

### PR DESCRIPTION
**Pull request by zhzhuang-zju**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Runs Trivy as GitHub action to scan Docker container image for vulnerabilities. 

We talked about this task at [Community Meeting 2023-01-17](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit?pli=1#heading=h.yrs5d1w97tdg).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Please see https://github.com/zhzhuang-zju/karmada/actions/runs/6530626808/job/17730271059 for more specific effects.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by zhzhuang-zju** [comment]
This pr add image-scanning CI，providing the ability to scan the master branch image during push and scheduled old version image scanning.
- scan the master branch image during push
  When the code is merged into the master branch, the image scan is started. CI fails when a vulnerability is detected.

image

 You can view the scan results of a single image in each job.

image

 Or view the overall results in the red box area.

image

- scheduled old version image scanning
  This workflow runs every Saturday and every Sunday at 20:00 UTC to detect vulnerabilities in old versions(release-1.5, release-1.6, release-1.7) of images. CI fails when a vulnerability is detected.

image

You can view the scan results of each image in the green box, and the overall scan results of different versions of the images in the red box area.

File: .github/workflows/ci-image-scanning.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: image-scanning
+on:
+  push: 
+jobs:
+  use-trivy-to-scan-image:
+    name: image-scanning
+    if: ${{ github.repository == 'karmada-io/karmada' }}
+    runs-on: ubuntu-20.04
```
Comment (RainbowMango) [inline .github/workflows/ci-image-scanning.yaml:8]:
```suggestion
    runs-on: ubuntu-22.04
```
We are now using `ubuntu-22.04`.

File: .github/workflows/ci-image-scanning.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          make image-${{ matrix.target }}
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.12.0
+        with:
+          image-ref: 'docker.io/karmada/${{ matrix.target }}:latest'
+          format: 'table'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          exit-code: '1' 
+
```
Comment (RainbowMango) [inline .github/workflows/ci-image-scanning.yaml:40]: What's this?

Comment (RainbowMango) [reply .github/workflows/ci-image-scanning.yaml:40]: Looks like an extra blank line

Comment (XiShanYongYe-Chang) [reply .github/workflows/ci-image-scanning.yaml:40]: A blank line without indents and spaces is required

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**